web/src/pages/loading.tsx

```diff
@@ -43,18 +43,18 @@ const LoadingPage = (): ReactElement => {
     if (!updateQueue) return;
     const business = updateQueue.currentBusiness();
     if (!onboardingCompleted(business)) {
-      router.push(ROUTES.onboarding);
+      router && router.push(ROUTES.onboarding);
     } else if (business.preferences.returnToLink) {
       const pageLink = business.preferences.returnToLink;
       updateQueue
         .queuePreferences({ returnToLink: "" })
         .update()
         .then(() => {
-          router.push(pageLink);
+          router && router.push(pageLink);
         });
     } else {
       updateQueue.update().then(() => {
-        router.push(ROUTES.dashboard);
+        router && router.push(ROUTES.dashboard);
       });
     }
   }, userData);
```

Comment:
could this one just be `userData && router` as the second argument to `useMountEffectWhenDefined` instead of the other checks

Reply:
useMountEffectWhenDefined can only take one dependency variable, adding an array with two variables would cause issues. There is a ticket to remove useMountEffectWhenDefined because it leads to pages not reloading.  I think we can address it there.